sp-sim/src/gimlet.rs

```diff
@@ -115,43 +148,44 @@ impl Gimlet {
-            gimlet.common.bind_addrs
-        {
-            // bind to our two local "KSZ" ports
-            assert_eq!(bind_addrs.len(), 2); // gimlet SP always has 2 ports
-            let servers = future::try_join(
-                UdpServer::new(
-                    bind_addrs[0],
-                    gimlet.common.multicast_addr,
-                    &log,
-                ),
-                UdpServer::new(
-                    bind_addrs[1],
-                    gimlet.common.multicast_addr,
-                    &log,
-                ),
-            )
-            .await?;
-            let servers = [servers.0, servers.1];
-
-            for component_config in &gimlet.common.components {
-                let id = component_config.id.as_str();
-                let component = SpComponent::try_from(id)
-                    .map_err(|_| anyhow!("component id {:?} too long", id))?;
-
-                if let Some(addr) = component_config.serial_console {
-                    let listener =
-                        TcpListener::bind(addr).await.with_context(|| {
-                            format!("failed to bind to {}", addr)
+        let (local_addrs, handler, responses_sent_count) =
+            if let Some(bind_addrs) = gimlet.common.bind_addrs {
+                // bind to our two local "KSZ" ports
+                assert_eq!(bind_addrs.len(), 2); // gimlet SP always has 2 ports
+                let servers = future::try_join(
+                    UdpServer::new(
+                        bind_addrs[0],
+                        gimlet.common.multicast_addr,
+                        &log,
+                    ),
+                    UdpServer::new(
+                        bind_addrs[1],
+                        gimlet.common.multicast_addr,
+                        &log,
+                    ),
+                )
+                .await?;
+                let servers = [servers.0, servers.1];
+
+                for component_config in &gimlet.common.components {
+                    let id = component_config.id.as_str();
+                    let component =
+                        SpComponent::try_from(id).map_err(|_| {
+                            anyhow!("component id {:?} too long", id)
                         })?;
-                    info!(
-                        log, "bound fake serial console to TCP port";
-                        "addr" => %addr,
-                        "component" => ?component,
-                    );
 
-                    serial_console_addrs.insert(
+                    if let Some(addr) = component_config.serial_console {
+                        let listener =
+                            TcpListener::bind(addr).await.with_context(
+                                || format!("failed to bind to {}", addr),
+                            )?;
+                        info!(
+                            log, "bound fake serial console to TCP port";
+                            "addr" => %addr,
```

Comment:
(can't put this comment in the right place -- thanks GitHub)

Lines 189-170 look incorrectly aligned.